example_unmarshaler_test.go

```diff
@@ -0,0 +1,74 @@
+package plist_test
+
+import (
+	"fmt"
+
+	"github.com/groob/plist"
+)
+
+const data = `<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+	<key>typekey</key>
+	<string>A</string>
+	<key>typeAkey</key>
+	<string>VALUE-A</string>
+</dict>
+</plist>`
+
+type TypeDecider struct {
+	ActualType interface{} `plist:"-"`
+}
+
+type TypeA struct {
+	TypeAKey string `plist:"typeAkey"`
+}
+
+type TypeB struct {
+	TypeBKey string `plist:"typeBkey"`
+}
+
+func (t *TypeDecider) UnmarshalPlist(f func(interface{}) error) error {
+	// stub struct for decoding a single key to tell which
+	// specific type we should umarshal into
+	typeKey := &struct {
+		TypeKey string `plist:"typekey"`
+	}{}
+	if err := f(typeKey); err != nil {
+		return err
+	}
+
+	// switch using the decoded value to determine the correct type
+	switch typeKey.TypeKey {
+	case "A":
+		t.ActualType = new(TypeA)
+	case "B":
+		t.ActualType = new(TypeB)
+	case "":
+		return fmt.Errorf("empty typekey (or wrong input data)")
+	default:
+		return fmt.Errorf("unknown typekey: %s", typeKey.TypeKey)
+	}
+
+	// decode into the actual type
+	return f(t.ActualType)
+}
+
+// Example_unmarshaler demonstrates using structs that use the Unmarshaler interface.
+func Example_unmarshaler() {
```

Comment:
If this is renamed ExampleUnmarshaler, it will be come an example on the interface instead of a package-level example.

Reply:
Huh. That was my goal! I tried to find examples and docs of how to do that, but came up empty. Was that documented somewhere?

Reply:
This blog post is what I was going off of: https://go.dev/blog/examples